nbody_gravitational/NBodyGravitational.jl

```diff
@@ -0,0 +1,173 @@
+module NBodyGravitational
+
+using DiffEqBase, DifferentialEquations, StaticArrays, Plots, StaticArrays
+
+
+grav_const = 6.673e-11
+
+#=
+    Represents a body/particle in an N-body gravitational problem
+    m - mass
+    r - initial position
+    v - initial velocity
+=#
+struct MassBody{mType, cType}
+    m :: mType
+    r :: Vector{cType}
+    v :: Vector{cType}
+    
+    function MassBody{mType,cType}(m::mType, r::Vector{cType}, v::Vector{cType})
+        @assert length(r)==length(v) "The length of the position (r) and velocity (v) components should match"
+        new(m,r,v)
+    end
+end
+
+MassBody(m::mType, r :: Vector{cType}, v :: Vector{cType}) where {mType <: AbstractFloat, cType  <: AbstractFloat} = MassBody{mType, cType}(m,r,v)
+
+#=
+    Contains the necessary information for solving an N-body gravitational problem
+    bodies - interacting bodies/particles
+    tSpan - time, for which we want to evaluate the solution
+    G - gravitational constant
+=#
+struct NBodyGravProblem
+    bodies :: Array{MassBody, 1}
+    tSpan :: Tuple{Float64, Float64}
+    G :: Float64
+end
+
+NBodyGravProblem(bodies, tSpan::Tuple{Float64, Float64}) = NBodyGravProblem(bodies, tSpan, grav_const)
+
+#= 
+    As a quick approach to solving an N-body problem we will just 
+    transform NBodyGravProblem to an ODEProblem
+
+    Here u will be an array consisting of a consecutive positional coordinates r 
+    following by components of velocities v
+=#
+import Base.convert
+convert(::Type{ODEProblem}, x:: NBodyGravProblem) =
+begin
+    n = length(x.bodies)
+    u0 = zeros(6*n);
+    m = zeros(n)
+    
+    for i=1:n
+        ith = 3(i-1)+1;        
+        u0[ith:ith+2] = x.bodies[i].r;
+        u0[3n+ith:3n+ith+2] = x.bodies[i].v 
+        m[i] = x.bodies[i].m
+    end    
+    
+    function ode_system!(du, u, p, t)
+        du[1:3n] = @view u[3n+1:6n];
+        for i=1:n
+            ith = 3(i-1)+1;
+            ri = @SVector [u[ith] , u[ith+1], u[ith+2]]
+            du[3n+ith:3n+ith+2] = 
+            begin
+                accel = @SVector [0.0, 0.0, 0.0]; 
```

Comment:
The syntax is a little odd. Just calculate accel and then write the to du?

Reply:
Changed it.